lib/create-options.js

```diff
@@ -10,9 +11,16 @@ function createOptions(cwd, context) {
     if (context.options.gid) options.gid = context.options.gid;
   }
   options.env = Object.create(process.env);
-  options.env['npm_loglevel'] = 'error';
+  options.env['npm_loglevel'] = context.options.npmLevel;
   options.env['npm_config_tmp'] = context.npmConfigTmp;
 
+  if (context.options.nodedir) {
+    const nodedir = path.resolve(process.cwd(), context.options.nodedir);
+    options.env['npm_config_nodedir'] = nodedir;
+    context.emit('data', 'verbose', `${context.module.name} nodedir`,
```

Comment:
The fixes were being applied after the env var was set, because `--nodedir` was passed as a command line option. As we're not doing that they shouldn't be necessary. They're still not needed for the env var, and they shouldn't be needed for the `context.emit()`.